### PR TITLE
Adds the initial implementation of the drop metadata class

### PR DESCRIPTION
**Pull request by matthewbentley**
Still todo: comments, write to / read from files, create drop metadata
from a directory, interface with file metadata

Publishing for initial review (comments coming later today)

### CODE REVIEW

File: syncr_backend/drop_metadata.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        expected = self._gen_files_hash()
+        if given != expected:
+            raise Exception("Invalid files hash")
+
+    @property
+    def unsigned_header(self) -> Dict[str, Any]:  # TODO: type this better?
+        """Get the unsigned version of the header
+        The signature is set to b"", and the files list is {}
+
+        :retunrs: A dict that is the drop metadata header, without a signature
```
Comment (bretttjohnson1) [inline syncr_backend/drop_metadata.py]: returns typo

File: syncr_backend/drop_metadata.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        self.owner = primary_owner
+        self.other_owners = other_owners
+        self.signed_by = signed_by
+        self.files = files
+        self.sig = sig
+        self._protocol_version = protocol_version
+        self._files_hash = files_hash
+
+    @property
+    def files_hash(self) -> bytes:
```
Comment (DLance96) [inline syncr_backend/drop_metadata.py]: Docstring

File: syncr_backend/drop_metadata.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    :param path: The directory to make metadata from
+    :param name: The name of the drop to create
+    :param drop_id: The drop id of the drop metadata, must match the owner
+    :param owner: The owner, must match the drop id
+    :param other_owners: Other owners, may be empty
+    :return: A tuple of the drop metadata, and a dict from file names to file
+    metadata
+    """
+    files = {}
+    for (dirpath, dirnames, filenames) in os.walk(path):
```
Comment (DLance96) [inline syncr_backend/drop_metadata.py]: Pull this out into its own method

Comment (DLance96) [inline syncr_backend/drop_metadata.py]: Pull out into its own method

**Pull request merged**